packages/runtime/container-runtime/src/test/blobManager.spec.ts

```diff
@@ -181,7 +181,11 @@ export class MockRuntime
 			this.processBlobsP.reject(new Error("fake error"));
 		}
 		this.processBlobsP = new Deferred<void>();
-		await Promise.all(blobPs);
+		for (const blobP of blobPs) {
+			// allow uploadBlob to finish their then/catch statements
+			await blobP.then().catch(() => {});
```

Comment:
I'm not sure allSettled does all the trick here but I guess I can just manually wait all blobs. I guess I can add other tests to ensure proper behavior when uploading multiple blobs.

Reply:
There are already tests uploading multiple blobs and now there is one uploading multiple failing blobs.

Reply:
```suggestion
			await blobP.catch(() => {});
```

Reply:
allSettled should do most of what you want. i don't know how it exits, so it might have to be `Promise.allSettled(blobPs).catch(()=>{});`